macOS: add support for right side modifier keys

Koston-0xDEADBEEF opened the pull request:
Notes:
 * Tested with FreeBSD server & macOS client
 * Function `handleModifierKeys()` is untested; I was unable to trigger a breakpoint in it. Maybe it's only hit when running a server?
 
 There's still something a bit off in the logic when handling multiple simultaneous modifier key presses: Pressing Alt_R-Shift_L works, but pressing Shift_L-Alt_R causes latter to get interpreted with incorrect keycode. I'm not sure if I have time/energy left to address it, but I'll try.
Edit: This seems to be an unrelated XWindow server side issue.
Edit2: It's not a bug, just what happens when shift modifier is applied to alt_r with my current configuration

Koston-0xDEADBEEF commented:
Code is now tested both as client and server and added support for distinguishing right side modifiers also as server.

p12tic commented on src/lib/platform/OSXKeyState.cpp:
```diff
@@ -108,22 +97,21 @@ static const KeyEntry    s_controlKeys[] = {
-    { kKeyMeta_L,        s_superVK },
-    { kKeyMeta_R,        s_superVK }, // 61
+    // modifier keys. Support for right side keys is a later addition in macOS.
+    { kKeyShift_L,          kVK_Shift },        // 56
```
Let's align with the comma (`{ kKeyShift_L, kVK_Shift },`), because otherwise all lines will need to be modified whenever a very long first work appears in the list.

Koston-0xDEADBEEF replied:
fixed

p12tic commented:
@Koston-0xDEADBEEF Thanks a lot for the PR. It looks good, I only had a single nitpick regarding alignment.

Koston-0xDEADBEEF commented:
> @Koston-0xDEADBEEF Thanks a lot for the PR. It looks good, I only had a single nitpick regarding alignment.

Cheers - I fixed the alignment as requested.

shymega commented:
@Koston-0xDEADBEEF All looks good to me. One more thing, could you rebase your PR to squash the two commits into one? That way we don't have a dangling 'fix alignment' commit. Once that's done, either @p12tic or I can merge. Cheers.

Koston-0xDEADBEEF commented:
> @Koston-0xDEADBEEF All looks good to me. One more thing, could you rebase your PR to squash the two commits into one? That way we don't have a dangling 'fix alignment' commit. Once that's done, either @p12tic or I can merge. Cheers.

I've squashed the commits and rebased against upstream.

shymega commented:
OK, I'll merge it now.

shymega commented:
Thank you for your work on this, @Koston-0xDEADBEEF, very grateful for your effort. :)

Koston-0xDEADBEEF commented:
> Thank you for your work on this, @Koston-0xDEADBEEF, very grateful for your effort. :)

Happy to help. I will see if I can also fix some memory corruption bugs while at it.
